gtkmm-plplot/plot2d.cpp

```diff
@@ -184,6 +187,8 @@ void Plot2D::set_axis_logarithmic_x(bool _log10) {
 void Plot2D::set_axis_logarithmic_y(bool _log10) {
   //need to check that all values are positive!
   if (_log10) {
+    if (get_axis_logarithmic_x())
```

Comment:
I don't understand why you have modified `Plot2D::set_axis_logarithmic_y` at all if you're only supporting datetimes on the X-axis (for now?)

Perhaps something like this should go into `Plot2D::set_axis_logarithmic_x`? But even then you would need to check if datetimes are going to be used before throwing an exception.